test/integration/controller/core/workload_controller_test.go

```diff
@@ -180,7 +183,7 @@ var _ = ginkgo.Describe("Workload controller", func() {
 				Queue(localQueue.Name).
 				Request(corev1.ResourceCPU, "1").
 				Admit(testing.MakeAdmission(clusterQueue.Name).
-					Flavor(corev1.ResourceCPU, flavorOnDemand).Obj()).
+					Flavor(corev1.ResourceCPU, flavorSpot).Obj()).
```

Comment:
We need to use other resourceFlavors, not `flavorOnDemand`, to avoid conflict with different test cases on workload_controller_test.

Reply:
No needed for the reasons same as https://github.com/kubernetes-sigs/kueue/pull/415#discussion_r1051645138.